File: ShoeStore/ShoeStore.Domain/Entities/CartItem.cs
using ShoeStore.Domain.Entities;

public class CartItem
{
    public long Id { get; set; }
    public long CartId { get; set; }
    public int ProductId { get; set; }
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }

    public Cart Cart { get; set; } = null!;
    public Product Product { get; set; } = null!;
}

File: shoestorefe/lib/domain/entities/cart_item.dart
class CartItem {
  final int id;
  final int? cartId; // Nullable since API may not return it
  final int productId;
  final int quantity;
  final double unitPrice;
  CartItem({
    required this.id,
    this.cartId, // Optional
    required this.productId,
    required this.quantity,
    required this.unitPrice,
  });
}
